src/lightning_app/core/queues.py

```diff
@@ -234,6 +234,16 @@ def __init__(
         self.redis = redis.Redis(host=host, port=port, password=password)
 
     def put(self, item: Any) -> None:
+        from lightning_app import LightningWork
+
+        is_work = isinstance(item, LightningWork)
+
+        # The backend isn't picklable
+        # Raises a TypeError: cannot pickle '_thread.RLock' object
+        if is_work:
+            backend = item._backend
```

Comment:
Will this create race conditions? For a brief period of time `item` will not have a `_backend` attached and we cannot know how it is used in eventual parallel threads. Modifying `item` in-place this way is not ideal.

Why do we need to pickle the whole work BTW? Can't we just pickle the arguments to the run method?

Reply:
This is how dynamic work is supported and one of its limitations. We are pickling the entire work and receiving it on the other side.

Right now, there isn't any thread accessing the work, so there won't be any issues.

Reply:
I know we talked about not pickling the entire thing multiple times. Can you add a TODO here? It's not super urgent but we'll need to address this sooner or later.